### ISSUE TITLE
Bad probability distribution after classification training

### ISSUE THREAD

**Issue by underthesand** [ultralytics/ultralytics#8019]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Predict

### Bug

Hello, I would like to improve my classification model, I tried several parameters, balancing the dataset, it doesn't change anything. 

I have 10 classes, the predictions are good when the model focuses on one class (score greater than 0.90). But for my usage, in many cases I have to predict on images very close to images present in the training dataset of all classes at the same times and in this case we could expect the model to distribute the probability over all of the classes but instead it only targets four classes (always the same), these four classes have a score between 0.15 and 0.30 and the other six classes have a score very close to 0. The expected result should be a score of 0.1 approximately in each class.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Burhan-Q** [comment]
@fcakyon correct me if I'm wrong, but I believe the current implementation of the classification model will only provide prediction for a _single class_ per image. @underthesand this means that if you'll be predicting for images which _may_ have multiple instances of difference classes, you'll need to generate additional labels for these scenarios. As an example:

```
0: c1
1: c2
2: c3
3: c1+c2
4: c1+c3
5: c2+c3
6: c1+c2+c3
...
```

Meaning you'll also need to ensure your dataset has sufficient instances of not only each individual class instance, but also the combination of classes. Additionally, if there are cases that would _never_ exist, then these examples do not need to be included in your training dataset, but I would advise that you're **certain** this will be always be the case or is an acceptable assumption to make.

**Comment by fcakyon** [comment]
@Burhan-Q @underthesand ultralytics classification training does not support multi-class multi-label classification. It supports multi-class single-label classification. This means there is a softmax before the output, and prediction probabilities sum up to 1. Thus, it is expected to have predictions close to zero (due to the softmax function's exponential nature).

**Comment by underthesand** [comment]
@fcakyon thank you. Indeed, I think the problem lies at this level. It seems that the model's predictions are skewed by the fact that the total must equal 1.
To reproduce this, just include the same image in all classes in the training dataset. Inferring this image with the obtained model leads to an uninterpretable result, whereas it should give a similar score for each class: 1/number of classes. Unfortunately, I cannot add combined classes like c1+c2+... because when constituting the dataset, I don't know if the image can belong to more than one class. I was indeed thinking of using the model as a measure of similarity by isolating images that return similar scores across multiple classes.

Do you think it's possible to replace the softmax function with something else, or should I use another tool?
